File: SortingAlgorithm/insertionsort.js
let sort=(arr)=>{

for(let i=0;i<arr.length-1;i++){
       let temp=0;
    for(let j=i+1;j<arr.length;j++){
        if(arr[i]>arr[j]){
           temp=arr[i];
           arr[i]=arr[j];
           arr[j]=temp;
        }
    }
}
return arr;
}


let arr=[5,4,3,2,1];

console.log(sort(arr));

File: dsapractice/algorithms/InsertionSort.java
package algorithms;

import java.util.Arrays;

public class InsertionSort {
    
  static int[] insertionSort(int[] arr){


int temp,j;

for(int i=1;i<arr.length;i++){
    temp=arr[i];
    j=i;
    while(j>0&&arr[j-1]>temp){
      arr[j]=arr[j-1];
      j=j-1;
    }
  arr[j]=temp;
}

    return arr;
  }


public static void main(String[] args) {
    
int arr[]={36,19,29,12,5};

System.out.println(Arrays.toString(insertionSort(arr)));

}

}
